enhancements/network/user-defined-network-segmentation.md

```diff
@@ -379,11 +382,260 @@ any user-defined primary network.
+```golang
+// UserDefinedNetwork describes OVN network request for a Namespace.
+type UserDefinedNetwork struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="Spec is immutable"
+	// +kubebuilder:validation:XValidation:rule="(self.topology == 'Layer3' && size(self.subnets) > 0) || self.topology != 'Layer3'", message="Subnets is required for Layer3 topology"
+	// +required
+	Spec UserDefinedNetworkSpec `json:"spec"`
+	// +optional
+	Status UserDefinedNetworkStatus `json:"status,omitempty"`
+}
+```
+Suggested API validation rules:
+- Spec defined as mutable 
+  Avoid incomplete state in a scenario where a NAD is created according to UDN spec, and pods connected to the network, 
+  UDN spec changes, now pods connected to a network that was created from previous revision spec.
+- When `Topology: Layer3` is specified, `Subnets` must be provided.
+
+```golang
+// UserDefinedNetworkSpec defines the desired state of UserDefinedNetwork.
+type UserDefinedNetworkSpec struct {
+	// The topological configuration for the network.
+	// +kubebuilder:validation:Enum=Layer2,Layer3
+	Topology NetworkTopology `json:"topology"`
+
+	// The network role in the pod (e.g.: Primary, Secondary).
+	// +kubebuilder:validation:Enum=Primary,Secondary
+	Role NetworkRole `json:"role"`
+
+	// The maximum transmission unit (MTU).
+	// The default value is 1400.
+	// +optional
+	Mtu uint `json:"mtu,omitempty"`
+
+	// The subnet to use for the network across the cluster.
+	// Only include the CIDR for the node. E.g. 10.100.200.0/24.
+	//
+	// IPv6 (2001:DBB::/64) and dual-stack (192.168.100.0/24,2001:DBB::/64) subnets are supported.
+	//
+	// When omitted, the logical switch implementing the network only provides layer 2 communication,
+	// and users must configure IP addresses for the pods.
+	// Port security only prevents MAC spoofing.
+	// +optional
+	Subnets []string `json:"subnets,omitempty"`
+
+	// A list of CIDRs.
+	// IP addresses are removed from the assignable IP address pool and are never passed to the pods.
+	// +optional
+	ExcludeSubnets []string `json:"excludeSubnets,omitempty"`
+
+	// Subnet used inside the OVN network topology.
+	// E.g. 100.65.0.0/24. IPv6 (fd99::/64) and dual-stack (100.65.0.0/24,fd99::/64).
+	// When omitted and network `role` is `primary` the following default CIDR is used `100.65.0.0/24`, IPv6 `fd99::/64`.
+	// +optional
+	JoinSubnets []string `json:"joinSubnets,omitempty"`
+
+	// Control IP addresses management strategy. 
+	// When `Persistent` is specified it enable workloads have persistent IP addresses.
+	// For example: Virtual Machines will have the same IP addresses along their lifecycle (stop, start migration, reboots).
+	// Supported by Topology Layer2 only.
+	// +kubebuilder:validation:Enum=Enable,Disable
+	// +optional
+	IPAMStrategy string `json:"ipamStrategy,omitempty"`
```

Comment:
Still think `lifecycle` is much more informative than strategy that is too generic.
```suggestion
	IPAMLifecycle string `json:"ipamLifecycle,omitempty"`
```

Reply:
Done